tasks/pre_checks/validate_data_center_name.yml

```diff
@@ -0,0 +1,14 @@
+- name: Validate Data Center name format
+  block:
+  - name: Fail if Data Center name format is incorrect
+    fail:
+      msg: >-
+        "Invalid Data Center name format. Data Center name may only contain letters, numbers, '-', or '_'."
+        " Got {{ he_data_center }}"
+    when: not he_data_center | regex_search( "^[a-zA-Z0-9_-]+$" )
+  - name: Validate Cluster name
+    fail:
+      msg: >-
+        "Cluster name cannot be 'Default'. This is a reserved name for the default DataCenter. Please choose"
+        " another name for the cluster"
+    when: he_data_center != "Default" and he_cluster == "Default"
```

Comment:
IMHO we don't need to fail on this case

Reply:
why not ? 
if the cluster name is 'Default' and the data Center name is not 'Default' the deployment will fail. so why wait until the middle of the deployment and not fail at the beginning ?

Reply:
I confused and thought that it's valid but you right.